qdev_wrappers/qdev_fitter.py

```diff
@@ -0,0 +1,141 @@
+    str_id = '{0:03d}'.format(id)
+    if samplefolder==None:
+        check_experiment_is_initialized()
+
+        path = qc.DataSet.location_provider.fmt.format(counter=str_id)
+        data = qc.load_data(path)
+    else:
+        path = '{}{}{}'.format(samplefolder,sep,str_id)
+        data = qc.load_data(path)
+
+    # Get dataname to be fitted
+    if dataname==None:
+        # Function only made for a single dataname. Check there isnt more datanames in the dataset.
+        key_data = [key for key in data.arrays.keys() if "_set" not in key[-4:]]
+        if len(key_data) != 1:
+            raise RuntimeError('Dataset has more than one dataname. Choose dataname to be fitted from: {}.'.format(', '.join(key_data)))
+    else:
+        # Check input dataname is in dataset.
+        if dataname not in [key for key in data.arrays.keys() if "_set" not in key[-4:]]:
+            raise RuntimeError('Dataname not in dataset. Input dataname was: \'{}\' while dataname(s) in dataset are: {}.'.format(dataname,', '.join(key_data)))
+        key_data = [dataname]
+
+    # Get handle for fitting function
+    if isinstance(fitfunction_in,str):
+        if fitfunction_in not in list(globals().keys()):
+            raise RuntimeError('Fitfunction not defined.')
+        fitfunction = globals()['{}'.format(fitfunction_in)]
+    elif isinstance(fitfunction_in,types.FunctionType):
+        fitfunction = fitfunction_in
+    else:
+        print('Invalid input. Only takes strings of predefined fucntions or function input.')
+
+
+    # Do fits separately for 1D and 2D datasets
+    keys_set = [key for key in data.arrays.keys() if "_set" in key[-4:]]
+
+    # Do fit and plot for 1D data
+    if len(keys_set) == 1:
+        xdata = getattr(getattr(data, keys_set[0]), 'ndarray')
+        ydata = getattr(getattr(data, key_data[0]), 'ndarray')
+        # Get initial guess on parameter is guess function is defined
+        if (p0==None and '{}_guess'.format(fitfunction_in) in list(globals().keys())):
+            p0 = globals()['{}_guess'.format(fitfunction_in)](xdata,ydata)
+        popt, pcov = popt, pcov = curve_fit(fitfunction, xdata, ydata, p0=p0, **kwargs)
+
+        if do_plots:
+            plot = plot_1D(getattr(data, keys_set[0]),getattr(data, key_data[0]),fitfunction,popt)
+            title_list = plot.get_default_title().split(sep)
+            title_list.insert(-1, 'analysis')
+            title = sep.join(title_list)
+            plt.savefig("{}_{}.png".format(title,fitfunction.__name__),dpi=500)
+            return popt, pcov, plot
+        else:
+            return popt, pcov
+
+    # Not implemented
+    if len(keys_set) == 2:
+        if xname==None:
+            raise RuntimeError('Need fitting dimension for 2D datasets. Choose fit dimension from: {}.'.format(', '.join(keys_set)))
+        raise RuntimeError('Fitting not ready for 2D datasets.')
+
+
+def check_experiment_is_initialized():
```

Comment:
This function is confusing. How is it different from 'check_experiment_is_initialized()' in show_num.py? If it is not we should rather import it, or move it to a different place and import it.

Reply:
its the same one. Just a copy as i didnt think to import it.